### PR TITLE
[KernelGen][Nvidia] Implement Triton-based _cudnn_rnn operator

### PR DESCRIPTION
**Pull request by yzw1128**
## Summary

Add a vendor-agnostic RNN forward pass using pure Triton kernels to replace the cuDNN-backed `aten::_cudnn_rnn` operator.

## Features

- **All RNN modes**: RNN_RELU, RNN_TANH, LSTM, GRU
- **Multi-layer and bidirectional**: arbitrary `num_layers` and `bidirectional=True`
- **LSTM projection**: `proj_size > 0` with correct `weight_stride0` (3 or 5)
- **Dropout during training**: `F.dropout` applied between layers
- **Packed variable-length sequences**: `batch_sizes` parameter fully supported

## Implementation

Six Triton kernels cover all RNN mode × sequence type combinations:

| Kernel | Mode | Sequence Type |
|--------|------|---------------|
| `_rnn_kernel` | RNN relu/tanh | uniform batch |
| `_rnn_packed_kernel` | RNN relu/tanh | packed |
| `_lstm_kernel` | LSTM (with/without projection) | uniform batch |
| `_lstm_packed_kernel` | LSTM (with/without projection) | packed |
| `_gru_kernel` | GRU | uniform batch |
| `_gru_packed_kernel` | GRU | packed |

Each kernel uses fp32 internal accumulation, double-buffered hidden state for chunk-level correctness, and `triton.language.sigmoid`/`tanh` for fused activation computation.

All computation paths are pure Triton — there are no PyTorch fallbacks.

## Accuracy

**223/224 tests pass** (dispatch test excluded due to FlagGems tensor wrapper incompatibility with internal `torch.sigmoid` calls during fp32 upcast).

Test coverage:
- 4 modes × 3 shapes × (1/2 layers) × (unidirectional/bidirectional) × (batch_first=False/True) × (fp16/fp32) = **192 accuracy tests**
- LSTM projection: 4 configs × (1/2 layers) × (unidirectional/bidirectional) × (fp16/fp32) = **16 tests**
- Dropout in training/eval modes: 5 + 4 = **9 tests**
- Packed sequences: 4 modes × (fp16/fp32) + 1 bidirectional = **9 tests**

Tolerance: `{torch.float32: 2e-3, torch.float16: 5e-3, torch.bfloat16: 3e-2}`

## Performance (H20, bfloat16)

| Shape | Kernel | cuDNN (ms) | FlagGems Triton (ms) | Speedup |
|-------|--------|-----------|---------------------|--------|
| RNN relu (16,4,128) | `_rnn_kernel` | 0.267 | 0.065 | **4.09x** |
| RNN tanh (16,4,128) | `_rnn_kernel` | 0.266 | 0.077 | **3.47x** |
| GRU (16,4,128) | `_gru_kernel` | 0.329 | 0.158 | **2.09x** |
| LSTM (16,4,128) 2-layer | `_lstm_kernel` | 0.761 | 0.430 | **1.77x** |
| LSTM batch_first (4,16,128) | `_lstm_kernel` | 0.388 | 0.220 | **1.76x** |
| LSTM bidir (16,4,128) | `_lstm_kernel` | 0.717 | 0.430 | **1.67x** |
| LSTM (16,4,128) 1-layer | `_lstm_kernel` | 0.359 | 0.222 | **1.61x** |
| LSTM (32,8,256) | `_lstm_kernel` | 0.533 | 1.551 | 0.34x |
| GRU (32,8,256) | `_gru_kernel` | 0.501 | 1.813 | 0.28x |

**Note**: Performance degrades for large hidden sizes (≥ 256) because the current kernels use a fixed `BLOCK_SIZE=64`. Future work will add `triton.autotune` to select optimal block sizes dynamically.

## Files Changed

```
 benchmark/test_cudnn_rnn.py    | 246 +++++++
 conf/operators.yaml            |  16 +
 src/flag_gems/__init__.py      |   1 +
 src/flag_gems/ops/__init__.py  |   2 +
 src/flag_gems/ops/cudnn_rnn.py | 1568 ++++++++++++++++++++++++++++++++++++++++
 tests/test_cudnn_rnn.py        | 638 ++++++++++++++++
 6 files changed, 2471 insertions(+)
```

### CODE REVIEW

**Comment by bin913** [comment]
@yzw1128 Could you please pay attention to the two test cases with poor performance?

File: conf/operators.yaml [hunk 1]
```diff
@@ -2314,6 +2314,22 @@ ops:
...
+      Vendor-agnostic RNN forward pass using Triton kernels that
+      replaces the cuDNN-backed _cudnn_rnn operator.  Supports
+      RNN_RELU, RNN_TANH, LSTM and GRU modes with multi-layer,
+      bidirectional, LSTM projection (proj_size > 0), dropout
+      during training, and packed variable-length sequences.
+    for:
+      - _cudnn_rnn
+    labels:
+      - aten
+      - NoCPU
```
Comment (Caeruleann) [inline conf/operators.yaml:2761]:
```suggestion
      - NoCPU
      - KernelGen
```

**Comment by bin913** [comment]
@yzw1128 Please resolve conflicts
